### ISSUE TITLE
3D training code and training data sets

### ISSUE THREAD

**Issue by wjtan99** [1adrianb/face-alignment-training#8]
Hi,  

After reading the code, I feel you only have the 2D FAN model in the code.  The 3D network is not in the model, did I miss anything?  Also when I looked at the 300W-LP directory, there are only 2d landmarks.   Where can I find the 3D training data sets?   

Thanks a lot. 
BTW, can you please not close the issue so quickly?  I may have follow up questions.

**Comment by wjtan99** [comment]
I found what is missing is the full 3D model with ResNet.  I also figured out the 3D landmark. But I still have no idea how the full 3D model, which is what I am most interested in, is trained.  Where is the ground truth of the depth from?

**Comment by wjtan99** [comment]
OK, I found all I need from the website http://www.cbsr.ia.ac.cn/users/xiangyuzhu/projects/3DDFA/main.htm.  
Thanks anyway,

**Comment by 1adrianb** [comment]
Hi @wjtan99 , sorry I was busy with some stuff. Yes the code is unfortuantelly missing this feature right now (training the ResNet for depth).
That been said I will try to take a weekend to release the rest of the training code seens I have to add and retrain a  few more models for the python version of the code anyway.

Yes, that is the right link with the data that I used for training. Let me know if you have any other questions.

**Comment by chenbohua3** [comment]
Hello Adrian:) Did you get the part missed previously? I would really like to learn how to training the ResNet for depth. Could you please help me with it? Sincerely. @1adrianb

**Comment by chenbohua3** [comment]
@wjtan99 could you tell me where i can found depth ground truth. i download 300W-LP, 300W-3D, 300w-3D-FACE from the website you provided, i can't find the  depth data for training 3D-full model. Thanks for your patience.

**Comment by ZYX-MLer** [comment]
@chenbohua3 I also met the same problem, do you know how to get the depth label?

**Comment by AndriiTsok** [comment]
@1adrianb  Hello Adrian! Would you mind helping us to integrate ResNetDepth training into FA training code? We have acquired a solid dataset of other body parts with depth, pose and anchor points, so we are eager to try your architecture to verify how it would operate with on the similar but completely different domain. 
Thank you.

**Comment by 1adrianb** [comment]
Hi @AndriiTsok and thanks a lot for your interest. Unfortunatelly at the moment I have no plans on updating this code, the instruction from the paper should be sufficient to reproduce the results. If something is unclear however, I am happy to offer a better explanation.
